### ISSUE TITLE
CSS View Transitions cross-document tests rely on navigating the top-level browsing context

### ISSUE THREAD

**Issue by gsnedders** [web-platform-tests/wpt#48825]
See, e.g., https://wpt.fyi/results/css/css-view-transitions/navigation/pagereveal-microtask-sequence.html?run_id=5170135061233664&run_id=5204002958737408&run_id=5075857710514176&run_id=5095519668142080

Chrome gives a TIMEOUT, Firefox returns "ERROR message: Document unloaded; maybe test navigated the top-level-browsing context?", Safari raises `webdriver.error.NoSuchFrameException` from within wptrunner.

Ultimately, we don't support navigating the top-level browsing context; for testharness tests, we have an assert where we expect things to fail if they even make it that far: https://github.com/web-platform-tests/wpt/blob/8be9ad55c941b65de8bc725fdc72eabca737ba43/tools/wptrunner/wptrunner/executors/base.py#L94

For reference: this assert goes all the way back to the very first commit which added wptrunner:
https://github.com/web-platform-tests/wpt/blob/4d16712a6eef81f996df259186e2343ec1a0275d/wptrunner/executor.py#L34-L35

For reftests, we don't seem to have anything explicitly checking for this, but this seems likely to be the cause of https://wpt.fyi/results/css/css-view-transitions/navigation/navigation-auto-excludes-reload.html?run_id=5170135061233664&run_id=5204002958737408&run_id=5075857710514176&run_id=5095519668142080 (which fails in Safari as we're waiting for the callback for `reftest-wait` being removed when the document gets navigated while we wait for the script to return; I'm somewhat surprised at that passing in Chrome — I wonder if there's WebDriver differences here?).

Admittedly, we don't seem to have this limitation documented anywhere, as far as I can see.

I expect this is the cause for most of the harness errors one can see on https://wpt.fyi/results/css/css-view-transitions/navigation?run_id=5170135061233664&run_id=5204002958737408&run_id=5075857710514176&run_id=5095519668142080, though I haven't investigated closely.

I don't know if the View Transitions cross-document tests could reasonably be rewritten to be navigating frames instead, or if this really is just blocked on coming up with some solution in WPT to handle the test page navigating (while handling the risk of somehow ending up with the results from the wrong page, which is not a purely hypothetical concern, as this has happened in various browser CI systems over the years).

cc @jgraham @jonathan-j-lee @WeizhongX on the wptrunner side
cc @noamr @mattwoodrow @vmpstr @khushalsagar on the view transitions side

(Somewhat prompted by https://github.com/web-platform-tests/interop/issues/771, to cross-reference the issues with the tests.)

**Comment by jonathan-j-lee** [comment]
Looks like #49738 fixes most of these tests. As of ~Dec 2024, Chromium CI runs new WPTs with wptrunner + WebDriver by default, instead of with a test-only `content_shell` browser that doesn't have this limitation. That will hopefully block nonportable tests from landing here.

I will leave this issue open for future doc/infra improvements.

**Comment by noamr** [comment]
> I will leave this issue open for future doc/infra improvements.

Is this necessary? Perhaps open a separate issue for that? It moggt be confusing to rely on an issue that is effectively resolved.

**Comment by gsnedders** [comment]
https://github.com/web-platform-tests/wpt/pull/49738 changed those using testharness.js, but not those that are reftests, right?
